src/SIL.Harmony/SnapshotWorker.cs

```diff
@@ -139,23 +139,14 @@ private async ValueTask ApplyCommitChanges(IEnumerable<Commit> commits, bool upd
     /// <param name="commit"></param>
     private async ValueTask MarkDeleted(Guid deletedEntityId, Commit commit)
     {
-        Expression<Func<ObjectSnapshot, bool>> predicateExpression =
-            snapshot => snapshot.References.Contains(deletedEntityId);
-        var predicate = predicateExpression.Compile();
-
-        var toRemoveRefFromIds = new HashSet<Guid>(await _crdtRepository.CurrentSnapshots()
-            .Where(predicateExpression)
+        var toRemoveRefFromIds = await GetSnapshotsReferencing(deletedEntityId, true)
             .Select(s => s.EntityId)
-            .ToArrayAsync());
-        //snapshots from the db might be out of date, we want to use the most up to date data in the worker as well
-        toRemoveRefFromIds.UnionWith(_rootSnapshots.Values.Where(predicate).Select(s => s.EntityId));
-        toRemoveRefFromIds.UnionWith(_pendingSnapshots.Values.Where(predicate).Select(s => s.EntityId));
+            .ToArrayAsync();
+
         foreach (var entityId in toRemoveRefFromIds)
         {
-            var snapshot = await GetSnapshot(entityId);
+            var snapshot = await GetSnapshot(entityId); // AsTracking
```

Comment:
this seems redundant now. We could just use GetSnapshotsReferencing and not select EntityId and we'd get out the whole snapshot and we can avoid querying it out again. Also since `.Select(s => s.EntityId)` is now on an AsyncEnumerable it's no longer projected back to the DB so we're already querying out the full object.